modules/indexing/src/main/java/org/apache/ignite/internal/visor/query/VisorQueryResetDetailMetricsTask.java

```diff
@@ -52,14 +53,7 @@ private VisorCacheResetQueryDetailMetricsJob(Void arg, boolean debug) {
 
         /** {@inheritDoc} */
         @Override protected Void run(Void arg) {
-            for (String cacheName : ignite.cacheNames()) {
-                IgniteCache cache = ignite.cache(cacheName);
-
-                if (cache == null)
-                    throw new IllegalStateException("Failed to find cache for name: " + cacheName);
-
-                cache.resetQueryDetailMetrics();
-            }
+            ((IgniteH2Indexing)ignite.context().query().getIndexing()).runningQueryManager().resetQueryHistoryMetrics();
```

Comment:
May be check for query().getIndexing() instanceOf IgniteH2Indexing for safety?

Reply:
Implemented check of indexing class.